lib/paperclip/io_adapters/abstract_adapter.rb

```diff
@@ -8,8 +8,20 @@ class AbstractAdapter
     delegate :binmode, :binmode?, :close, :close!, :closed?, :eof?, :path, :readbyte, :rewind, :unlink, :to => :@tempfile
     alias :length :size
 
+    def initialize(target, options = {})
+      @target = target
+      @options = options
+    end
+
     def fingerprint
-      @fingerprint ||= Digest::MD5.file(path).to_s
+      @fingerprint ||= begin
+        digest = @options.fetch(:hash_digest, Digest::MD5).new
```

Comment:
> I've removed the duplication from Paperclip::Attachment.

Let's remove this instance, and leave the one in the defaults hash for better code documentation. Thank you! 😃